tests/50federation/40publicroomlist.pl

```diff
@@ -0,0 +1,104 @@
+# Copied from 30rooms/70publicroomslist.pl, modified to test the federation publicRooms API
+test "Name/topic keys are correct",
+   requires => [ local_user_fixture(), $main::HOMESERVER_INFO[0], $main::HTTP_CLIENT ],
+
+   check => sub {
+      my ( $user, $info, $client ) = @_;
+
+      my $server_name = $info->server_name;
+
+      my %rooms = (
+         publicroomalias_no_name => {},
+         publicroomalias_with_name => {
+            name => "name_1",
+         },
+         publicroomalias_with_topic => {
+            topic => "topic_1",
+         },
+         publicroomalias_with_name_topic => {
+            name => "name_2",
+            topic => "topic_2",
+         },
+      );
+
+      Future->needs_all( map {
+         my $alias_local = $_;
+         my $room = $rooms{$alias_local};
+
+         matrix_create_room( $user,
+            visibility      => "public",
+            room_alias_name => $alias_local,
+            name            => $room->{name},
+            topic           => $room->{topic},
+         )
+      } keys %rooms )
+      ->then( sub {
+         $client->do_request_json(
+            method => "GET",
+            uri    => "/r0/publicRooms",
+            uri => "https://$server_name/_matrix/federation/v1/publicRooms",
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         log_if_fail "publicRooms", $body;
+
+         assert_json_keys( $body, qw( start end chunk ));
+         assert_json_list( $body->{chunk} );
+
+         my %seen = map {
+            $_ => 0,
+         } keys ( %rooms );
+
+         foreach my $room ( @{ $body->{chunk} } ) {
+            assert_json_keys( $room,
+               qw( world_readable guest_can_join num_joined_members )
+            );
+
+            my $name = $room->{name};
+            my $topic = $room->{topic};
+            my $canonical_alias = $room->{canonical_alias};
+
+            my $aliases = $room->{aliases};
+            if( not defined $aliases ) {
+               next;
+            }
+
+            foreach my $alias ( @{$aliases} ) {
+               foreach my $alias_local ( keys %rooms ) {
+                  $alias =~ m/^\Q#$alias_local:\E/ or next;
+
+                  my $room_config = $rooms{$alias_local};
+
+                  log_if_fail "Alias", $alias_local;
+                  log_if_fail "Room", $room;
+
+                  assert_eq( $canonical_alias, $alias, "Incorrect canonical_alias" );
```

Comment:
The message part here should be the name of the thing you were testing, so just

    assert_eq( $canonical_alias, $alias, "canonical alias" );

or somesuch